doc/plugins/eu.esdihumboldt.hale.doc.user/html/reference/cql_filter.html

```diff
@@ -58,6 +61,156 @@ <h2>Literals</h2>
 			<td>1.234</td>
 		</tr>
 	</table>
+	
+	<h2>Filter functions</h2>
+	<p>
+		For more complex filters you can use one or a combination of multiple functions. A selection of available functions is provided below. 
+		For the full list of functions, see GeoServer's <a href="http://docs.geoserver.org/latest/en/user/filter/function_reference.html">function reference</a>, 
+		but note that not all of the functions listed there are necessarily supported in a filter.
+	</p>
+	<p>
+		When defining a filter, replace all arguments with either a property name or a lexical expression 
+		(e.g. <tt>'John'</tt> or <tt>47</tt>). Also note, that all expressions must evaluate to true or false, so you may
+		have to add "<tt>= true</tt>" or "<tt>= false</tt>" after the function call, e.g. <tt>strMatches(name, 'Tom.*') = true<tt>.
```

Comment:
How does `strMatches(name, 'Tom.*') = true` differ from `strMatches(name, 'Tom.*')`? Is that some kind of special behavior in the filters?

Reply:
The difference is that Geotools compiles `strMatches(name, 'Tom.*') = true` to `IsEqualToImpl` (which implements `org.opengis.filter.Filter` => ok) but `strMatches(name, 'Tom.*')` to `FilterFunction_strMatches` (which is a `FunctionExpression` and not a `Filter`).

I extended `FilterGeoECqlImpl` so that it now transparently tries to turn an `Expression` into a
`Filter` by wrapping the `Expression` in a `PropertyIsEqualTo` (same as adding `= true` to the expression).